Mathlib/Analysis/SpecialFunctions/Pow/Complex.lean

```diff
@@ -42,6 +42,10 @@ theorem cpow_eq_zero_iff (x y : ℂ) : x ^ y = 0 ↔ x = 0 ∧ y ≠ 0 := by
   simp only [cpow_def]
   split_ifs <;> simp [*, exp_ne_zero]
 
+theorem cpow_ne_zero_iff {x y : ℂ} (hy : y ≠ 0) :
+    x ^ y ≠ 0 ↔ x ≠ 0 := by
+  simp only [ne_eq, cpow_eq_zero_iff, hy, not_false_eq_true, and_true]
```

Comment:
There's a new policy that terminal simps shouldn't be squeezed, so `simp [hy]`. I also wonder whether it is maybe better for `cpow_ne_zero_iff` to be precisely the opposite of `cpow_eq_zero_iff`, i.e. `x ^ y ≠ 0 ↔ x ≠ 0 ∨ y = 0`; if you really want this exact statement, it could be called `cpow_ne_zero_iff_of_exponent_ne_zero` (very wordy, I know).

Reply:
I implemented your suggestion

Reply:
Without squeezing the terminal simp :)